packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateRangeField.ts

```diff
@@ -84,36 +98,43 @@ export const useMultiInputDateRangeField = <TDate, TChildProps extends {}>({
     rangeValueManager.defaultErrorState,
   );
 
-  const startInputProps: UseDateFieldComponentProps<TDate, TChildProps> = {
+  const startFieldProps: UseDateFieldComponentProps<TDate, TTextFieldProps> = {
     error: !!validationError[0],
-    ...inStartTextFieldProps,
+    ...startTextFieldProps,
     disabled,
     readOnly,
     format,
+    unstableFieldRef: unstableStartFieldRef,
     value: valueProp === undefined ? undefined : valueProp[0],
     defaultValue: defaultValue === undefined ? undefined : defaultValue[0],
     onChange: handleStartDateChange,
+    selectedSections,
+    onSelectedSectionsChange,
   };
 
-  const endInputProps: UseDateFieldComponentProps<TDate, TChildProps> = {
+  const endFieldProps: UseDateFieldComponentProps<TDate, TTextFieldProps> = {
     error: !!validationError[1],
-    ...inEndTextFieldProps,
+    ...endTextFieldProps,
     format,
     disabled,
     readOnly,
+    unstableFieldRef: unstableEndFieldRef,
     value: valueProp === undefined ? undefined : valueProp[1],
     defaultValue: defaultValue === undefined ? undefined : defaultValue[1],
     onChange: handleEndDateChange,
+    selectedSections,
+    onSelectedSectionsChange,
   };
 
-  const startDateResponse = useDateField<TDate, TChildProps>({
-    props: startInputProps,
+  const startDateResponse = useDateField({
+    props: startFieldProps,
     inputRef: startInputRef,
-  });
-  const endDateResponse = useDateField<TDate, TChildProps>({
-    props: endInputProps,
+  }) as UseFieldResponse<TTextFieldProps>;
```

Comment:
![image](https://user-images.githubusercontent.com/3309670/225563450-1046ba07-c875-403d-894a-af9e9fd26cee.png)

I was not able to avoid this error :cry: